lock-api/src/main/java/com/palantir/lock/client/CommitTimestampGetter.java

```diff
@@ -16,96 +16,11 @@
-        DisruptorAutobatcher<Request, Long> autobatcher = Autobatchers.independent(consumer(leaseService, cache))
-                .safeLoggablePurpose("get-commit-timestamp")
-                .build();
-        return new CommitTimestampGetter(autobatcher);
-    }
-
-    public long getCommitTimestamp(long startTs, LockToken commitLocksToken) {
-        return AtlasFutures.getUnchecked(autobatcher.apply(ImmutableRequest.builder()
-                .startTs(startTs)
-                .commitLocksToken(commitLocksToken)
-                .build()));
-    }
-
-    @VisibleForTesting
-    static Consumer<List<BatchElement<Request, Long>>> consumer(
-            LockLeaseService leaseService, LockWatchEventCache cache) {
-        return batch -> {
-            int count = batch.size();
-            List<Long> commitTimestamps = new ArrayList<>();
-            while (commitTimestamps.size() < count) {
-                Optional<LockWatchVersion> requestedVersion = cache.lastKnownVersion();
-                GetCommitTimestampsResponse response =
-                        leaseService.getCommitTimestamps(requestedVersion, count - commitTimestamps.size());
-                commitTimestamps.addAll(process(batch.subList(commitTimestamps.size(), count), response, cache));
-                LockWatchLogUtility.logTransactionEvents(requestedVersion, response.getLockWatchUpdate());
-            }
-
-            for (int i = 0; i < count; i++) {
-                batch.get(i).result().set(commitTimestamps.get(i));
-            }
-        };
-    }
-
-    private static List<Long> process(
-            List<BatchElement<Request, Long>> requests,
-            GetCommitTimestampsResponse response,
-            LockWatchEventCache cache) {
-        List<Long> timestamps = LongStream.rangeClosed(response.getInclusiveLower(), response.getInclusiveUpper())
-                .boxed()
-                .collect(Collectors.toList());
-        List<TransactionUpdate> transactionUpdates = Streams.zip(
-                        timestamps.stream(),
-                        requests.stream(),
-                        (commitTs, batchElement) -> ImmutableTransactionUpdate.builder()
-                                .startTs(batchElement.argument().startTs())
-                                .commitTs(commitTs)
-                                .writesToken(batchElement.argument().commitLocksToken())
-                                .build())
-                .collect(Collectors.toList());
-        cache.processGetCommitTimestampsUpdate(transactionUpdates, response.getLockWatchUpdate());
-        return timestamps;
-    }
+public interface CommitTimestampGetter extends AutoCloseable {
+    long getCommitTimestamp(long startTs, LockToken commitLocksToken);
 
     @Override
-    public void close() {
-        autobatcher.close();
-    }
-
-    @Value.Immutable
-    interface Request {
-        long startTs();
-
-        LockToken commitLocksToken();
-    }
+    void close();
```

Comment:
Remove this method - you extend `AutoCloseable` anyway so any class that implements this interface will have to implement close.

Reply:
I wanted to kill exception thrown by AutoCloseable#close (which was the case before my refactor), here, the batcher only calls DisrupterAutobatcher#close which does the exception handling.

Reply:
You can anyway - if you just implement `close` in `BatchingCommitTimestampGetter` without `throws Exception` it just works.

Reply:
Discussed offline - we do this subtly differently internally, so was thinking of that. This works, or can equally extract a micro-interface that extends `AutoCloseable` but without the `throws Exception` on the close method.